service/src/main/java/bio/terra/cbas/runsets/monitoring/SmartRunsPoller.java

```diff
@@ -0,0 +1,85 @@
+package bio.terra.cbas.runsets.monitoring;
+
+import static java.util.stream.Collectors.groupingBy;
+
+import bio.terra.cbas.dao.RunDao;
+import bio.terra.cbas.dependencies.wes.CromwellService;
+import bio.terra.cbas.models.Run;
+import cromwell.client.ApiException;
+import cromwell.client.model.State;
+import java.time.OffsetDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SmartRunsPoller {
+
+  private final CromwellService cromwellService;
+  private final RunDao runDao;
+
+  private static final org.slf4j.Logger logger = LoggerFactory.getLogger(SmartRunsPoller.class);
+
+  public SmartRunsPoller(CromwellService cromwellService, RunDao runDao) {
+    this.cromwellService = cromwellService;
+    this.runDao = runDao;
+  }
+
+  public List<Run> updateRuns(List<Run> runs) {
+
+    // For metrics:
+    OffsetDateTime startTime = OffsetDateTime.now();
+
+    // Filter only updatable runs:
+    List<Run> updatableRuns = runs.stream().filter(Run::nonTerminal).toList();
+
+    // This has the nice outcome of counting even if the size is 0, which means this metric
+    // is created and stays current even if nothing is being updated:
+    logger.debug("METRIC: COUNT runs needing status update: {}", updatableRuns.size());
```

Comment:
TOL: Do we want to add a TODO here saying something like "convert this and below log messages into metrics when metrics framework has been added"?

Reply:
I was hoping the big `METRIC` tag would be sufficient?

Reply:
That seems fair 👍